querydsl-core/src/test/java/com/querydsl/core/util/MathUtilsTest.java

```diff
@@ -33,25 +35,42 @@ public void difference() {
     }
 
     @Test
-    public void cast() {
-        Integer value = 1;
-        assertEquals(BigDecimal.class, MathUtils.cast(value, BigDecimal.class).getClass());
-        assertEquals(BigInteger.class, MathUtils.cast(value, BigInteger.class).getClass());
-        assertEquals(Double.class, MathUtils.cast(value, Double.class).getClass());
-        assertEquals(Float.class, MathUtils.cast(value, Float.class).getClass());
-        assertEquals(Integer.class, MathUtils.cast(value, Integer.class).getClass());
-        assertEquals(Long.class, MathUtils.cast(value, Long.class).getClass());
-        assertEquals(Short.class, MathUtils.cast(value, Short.class).getClass());
-        assertEquals(Byte.class, MathUtils.cast(value, Byte.class).getClass());
-
-        assertEquals(BigDecimal.ONE, MathUtils.cast(value, BigDecimal.class));
-        assertEquals(BigInteger.ONE, MathUtils.cast(value, BigInteger.class));
-        assertEquals(Double.valueOf(1), MathUtils.cast(value, Double.class));
-        assertEquals(Float.valueOf(1), MathUtils.cast(value, Float.class));
-        assertEquals(Integer.valueOf(1), MathUtils.cast(value, Integer.class));
-        assertEquals(Long.valueOf(1), MathUtils.cast(value, Long.class));
-        assertEquals(Short.valueOf((short) 1), MathUtils.cast(value, Short.class));
-        assertEquals(Byte.valueOf((byte) 1), MathUtils.cast(value, Byte.class));
+    public void cast_returns_right_type() {
+        checkCast(1, BigDecimal.class);
+        checkCast(1, BigInteger.class);
+        checkCast(1, Double.class);
+        checkCast(1, Float.class);
+        checkCast(1, Integer.class);
+        checkCast(1, Long.class);
+        checkCast(1, Short.class);
+        checkCast(1, Byte.class);
+    }
+
+    @Test
+    public void cast_returns_argument_as_is_when_compatible() {
+        checkSame(BigDecimal.ONE, BigDecimal.class);
+        checkSame(BigInteger.ONE, BigInteger.class);
+        checkSame((double) 1, Double.class);
+        checkSame((float) 1, Float.class);
+        checkSame(1, Integer.class);
+        checkSame((long) 1, Long.class);
+        checkSame((short) 1, Short.class);
+        checkSame((byte) 1, Byte.class);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
```

Comment:
Consider `ExpectedException` and check the message. http://junit.org/javadoc/latest/org/junit/rules/ExpectedException.html